File: 78-subsets/subsets.java
class Solution {
    void helper(int[] nums, List<List<Integer>> result, List<Integer> curr, int start){
        result.add(new ArrayList<>(curr));
        for(int i = start ; i < nums.length; i++){
            curr.add(nums[i]);
            helper(nums, result, curr, i + 1);
            curr.remove(curr.size() - 1);
        }

    }
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> result = new ArrayList<>();
        List<Integer> curr = new ArrayList<>();
        helper(nums, result, curr, 0);
        return result;
    }
}

File: 78-subsets/subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        result = []
        def helper(nums, currlist, curr):
            nonlocal result
            result.append(currlist.copy())
            for i in range(curr, len(nums)):
                currlist.append(nums[i])
                helper(nums, currlist, i + 1)
                currlist.pop()
        helper(nums, [], 0)
        return result
        
        